branca/element.py

```diff
@@ -142,7 +151,25 @@ def add_child(
         name: Optional[str] = None,
         index: Optional[int] = None,
     ) -> "Element":
-        """Add a child."""
+        """Add a child and modify the child with a pointer to its parent."""
+
+        if child._parent is not None:
+            warnings.warn(
```

Comment:
I'm a bit conflicted about this warning. On the one hand, it's good to warn for potential issues. On the other hand, there are cases where you want to overwrite the `_parent` attribute, and generating warnings will lead to confusion. An example is the current workaround we have for using one Icon with multiple Markers: https://github.com/python-visualization/folium/pull/2068

Maybe as a compromise, instead of using the `warnings` library, instead use a logging message.

```
logger = logging.getLogger("branca")
logger.warn("bla"
```

What do you think?

Reply:
I'd also try to make the message shorter.